### ISSUE TITLE
Add PRAGMA foreign_keys = OFF/ON;

### ISSUE THREAD

**Issue by RandomFractals** [duckdb/duckdb#4201]
to enable creating tables without enforcing foreign key constraints when creating and initializing new duckdb database instance.

See SQLite documentation for more info:

https://www.sqlite.org/foreignkeys.html

https://www.sqlite.org/pragma.html#pragma_foreign_keys

**Comment by Tishj** [comment]
I'm a little confused after reading this.

To clarify, this would enable creating tables with FK constraints even when the referenced `table.column` does not satisfy the requirements of a FK constraint (table has to exist, column has to exist, column has to be a PRIMARY KEY) ?
Essentially changing the enforcing of these requirements to be lazy?

I could imagine this being useful when having to create a series of tables, knowing how they have to be linked, but not wanting to be forced into a specific creation order.
With the risk of being met with an error when attempting to scan the column with the FK constraint in a query, if those requirements are not met by then.

Are there other cases where this feature is useful?

**Comment by RandomFractals** [comment]
@Tishj yes, this feature request is mainly for creating new duckdb instance from other data sources without being concerned about the table dependencies order when tables are created and data is inserted.

I ran into this while trying to create duckdb instance from sqlite in this Observable notebook:

https://observablehq.com/@randomfractals/sqlite-to-duckdb?collection=@randomfractals/duckdb
